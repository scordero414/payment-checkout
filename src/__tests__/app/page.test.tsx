import MainPage from '@/app/page';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

jest.mock('@/redux/products/products-api', () => ({
  useGetProductsQuery: () => ({
    data: [
      {
        id: 2,
        title: 'newTitle',
        price: 500000,
        description:
          'Elevate your casual wardrobe with our Classic Red Pullover Hoodie. Crafted with a soft cotton blend for ultimate comfort, this vibrant red hoodie features a kangaroo pocket, adjustable drawstring hood, and ribbed cuffs for a snug fit. The timeless design ensures easy pairing with jeans or joggers for a relaxed yet stylish look, making it a versatile addition to your everyday attire.',
        images: 'https://fakestoreapi.com/img/61U7T1koQqL._AC_SX679_.jpg',
      },
    ],
    isLoading: false,
    error: null,
  }),
}));

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      push: () => null,
    };
  },
}));

describe('MainPage', () => {
  it('Renders the main page correctly', () => {
    const { container } = render(<MainPage />);
    expect(container).toMatchSnapshot();
  });
});
