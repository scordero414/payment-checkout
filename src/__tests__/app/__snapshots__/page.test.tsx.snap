// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Renders the main page correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-u8bj4b"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item css-1t8ikvu-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 MuiGrid-grid-xl-1.5 css-gz6hle-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-18lffqy-MuiPaper-root-MuiCard-root"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="product-image-2"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1bevvzh-MuiCardMedia-root"
                height="200"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-noWrap css-d5bee8-MuiTypography-root"
                >
                  newTitle
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  $500000
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
